examples/acados_matlab_octave/test/test_all_examples.m

```diff
@@ -29,105 +29,113 @@
+         %'../pendulum_on_cart_model/example_sim.m';
+         %'../pendulum_on_cart_model/example_solution_sens_closed_loop.m';  % Unable to resolve the name ocp.t_ocp.eval_param_sens.
+         '../pendulum_on_cart_model/experiment_dae_formulation.m';
+         '../race_cars/main.m';  % acados returns status 4
+         '../simple_dae_model/example_ocp.m';
+         %'../swarming/example_closed_loop.m';
+         '../swarming/example_ocp.m';
+         %'../swarming/example_sim.m';
+         %'../wind_turbine_nx6/example_closed_loop.m';
+         '../wind_turbine_nx6/example_ocp.m';
+         %'../wind_turbine_nx6/example_sim.m';
+%
+%         './test_checks.m';  % SHOULD FAIL, DOES
+%         './test_mstart_dirhe_lorentz.m';  % OK
+%         './test_ocp_OSQP.m';  % CRASH - solver not installed
+%         './test_ocp_linear_mass_spring.m';  % OK
+%         './test_ocp_pendulum_dae.m';  % OK
+%         './test_ocp_pendulum_on_cart.m';  % CRASH - solver not installed
+%         './test_ocp_qpdunes.m';  % CRASH - solver not installed
+%         './test_ocp_simple_dae.m';  % OK
+%         './test_ocp_wtnx6.m';  % FAIL - when not in installation folder?
+%         './test_sens_adj.m';  % OK
+%         './test_sens_forw.m';  % OK
+%         './test_sens_hess.m';  % OK
+%         './test_sim_dae.m';  % OK
+%         './test_target_selector.m'  % OK
+    };
+
+    pass = zeros(1, length(targets));  % keep track of test results
+    messages = cell(1, length(targets));  % and error messages
+    setenv("TEST_DIR", pwd)
+    for idx = 1:length(targets)
+        [dir, file, extension] = fileparts(targets{idx});
+
+        testpath = getenv("TEST_DIR");
+        save(strcat(testpath, "/test_workspace.mat"))
+        setenv("LD_RUN_PATH", strcat(testpath, "/", dir, "/c_generated_code"))
+
+        try
+            run(targets{idx});
+            test_val = true;
+            message = "";
+        catch exception
+            message = exception.message;
+            warning(exception.message);
+            clear exception
+            test_val = false;
+        end
 
-for test_number = 1:length(targets)
-    [target_dir,target_file,target_extension] = fileparts(targets{test_number});
-    cd(target_dir)
-    save test_workspace.mat  % due to "clear all" in the examples
-    try
-        run([target_file,target_extension])
-        clear; close all; clc;  % clean up after a succesful test
-        load test_workspace.mat
-        if contains(target_file,'simulink'); bdclose('all'); end
-        test_ok(test_number) = 1;  % save the result
-        disp(['testing ',targets{test_number},' succesful'])
-        pause(3)  % wait a bit before moving on
-    catch exception
-        clearvars -except exception;  % keep the caught error
+        % use absolute path, since current directory depends on point of failure
+        testpath = getenv("TEST_DIR");
+        load(strcat(testpath, "/test_workspace.mat"));
+        pause(3)
```

Comment:
why pause?

Reply:
retained it from the previous script, but removed the pause in the latest commit